aiohttp/client.py

```diff
@@ -210,6 +210,11 @@ class ClientTimeout:
 # https://www.rfc-editor.org/rfc/rfc9110#section-9.2.2
 IDEMPOTENT_METHODS = frozenset({"GET", "HEAD", "OPTIONS", "TRACE", "PUT", "DELETE"})
 HTTP_SCHEMA_SET = frozenset({"http", "https", ""})
+# special case ws and wss links when making requests as in some cases in asyncio
+# programs any other ways to request them might result in blocking the event loop
+# (or worse a deadlock).
+# Fixes https://github.com/aio-libs/aiohttp/issues/8481
+WS_SCHEMA_SET = frozenset({"ws", "wss"})
```

Comment:
Did some testing with this and it does seem to fix this issue `Jul 11 01:30:13 homeassistant homeassistant[569]: 2024-07-10 20:30:13.208 ERROR (MainThread) [aioharmony.hubconnector_websocket] 192.168.210.64: Exception trying to establish web socket connection for hub 15667900: ws://192.168.210.64:8088/?domain=svcs.myharmony.com&hubId=15667900
`

Reply:
if you want could you help me with making the tests for this change? I sorta got stuck trying to make them.

Reply:
I pushed a test for this.  It was a bit more work than I expected because everything was mocked out which made it a bit more difficult to get deep enough in the code to actually test the problem.

Someone else should take a look at the test for review purposes since it could use another set of 👀 for review

Reply:
Yeah, I guess the biggest issue is that there are no asserts, so it's not obvious what it actually testing.